src/libraries/System.Runtime.Numerics/src/System/Number.BigInteger.cs

```diff
@@ -503,6 +512,10 @@ private static ParsingStatus HexNumberToBigInteger(ref BigNumberBuffer number, o
                     ArrayPool<uint>.Shared.Return(arrayFromPool);
                 }
             }
+
+        FailExit:
```

Comment:
is using `goto` actually beneficial for the performance here?

Reply:
I don't actually remember where I take this pattern from.
The hex parsing itself can be rewritten with HexConverter in a follow-up PR. I will switch to simplest code.